src/libraries/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -527,87 +536,88 @@ private BigInteger(ReadOnlySpan<uint> value, bool negative)
         /// <param name="value"></param>
         private BigInteger(Span<uint> value)
```

Comment:
I'm curious why we have two separate implementations one for `Span<uint>` and one for `ReadOnlySpan<uint>` + isNegative. Could this ctor just do the checks to compute isNegative and then delegate to the other in order to avoid duplication? Or are they dealing with completely different formats?

Reply:
Mostly because `BigInteger` is an old type in desperate need of a rewrite 😆. I imagine we could, with a bit more refactoring, share some or most of the implementation here.

`private BigInteger(ReadOnlySpan<uint> value, bool negative)` is an optimized implementation that assumes we are already in almost the right shape. So `value` is the unsigned storage data (the absolute value) and we're really just trimming unnecessary leading zeros and deciding whether to store it in `_bits` or compress it into `_sign`.

`private BigInteger(Span<uint> value)` on the other hand is taking a two's complement little-endian span. So it has to detect if its positive or negative and if its negative fix it up to be the absolute value for storage purposes. So I expect we could implement it as:
* determine if negative, trimming leading sign data (`0xFF` if negative, `0x00` if positive)
* if negative, compute the two's complement (absolute value)
* call `private BigInteger(ReadOnlySpan<uint> value, bool negative)`